concent_api/core/tests/utils.py

```diff
@@ -197,13 +202,68 @@ def _test_database_objects(
         task_id                     = None,
         receive_delivered_status    = None,
     ):
-        self.assertEqual(StoredMessage.objects.last().type,           last_object_type.TYPE)
-        self.assertEqual(StoredMessage.objects.last().task_id,        task_id)
+        self.assertEqual(StoredMessage.objects.order_by('timestamp').last().type,      last_object_type.TYPE)
+        self.assertEqual(StoredMessage.objects.order_by('timestamp').last().task_id,   task_id)
 
         if receive_delivered_status is not None:
             self.assertEqual(ReceiveStatus.objects.last().delivered,        receive_delivered_status)
             self.assertEqual(ReceiveStatus.objects.last().message.task_id,  task_id)
 
+    def _test_subtask_state(
+        self,
+        task_id,
+        subtask_id,
+        subtask_state,
+        provider_key,
+        requestor_key,
+        list_of_nested_messages,
```

Comment:
It's misleading to call it a list when it's not required to be one. I've seen that you often pass a set here. I'd call it just `nested_messages`. Or `expected_messages`.

Reply:
Both names are already used somewhere else, so I change it to `expected_nested_messages`.